File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
            dict = defaultdict(int)
            for c in s:
                dict[c] += 1
            for c in t:
                dict[c] -= 1
            for v in dict.values():
                if v is not 0: return False
            return True

# Time complexity: O(n)
# Space complexity: O(n)

File: 0242-valid-anagram/0242-valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        if (s.length() != t.length()) return false;
        Map<Character, Integer> dict = new HashMap<>();
        for (char c : s.toCharArray()) {
            dict.put(c, dict.getOrDefault(c, 0) + 1);
        }
        for (char c : t.toCharArray()) {
            dict.put(c, dict.getOrDefault(c, 0) - 1);
        }
        for (int x : dict.values()) {
            if (x != 0) return false;
        }
        return true;
    }
}